### PR TITLE
Fix missing tags for remote client

### LINKED ISSUE #7775: No fuel available in the fuel list of stonesmelter and the stonesmelter cannot use fuel
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 1.0.128-ALHPA
- Structurize Version: 0.14.252-ALPHA
- Related Mods and their Versions:

## Expected behavior

In the previous version , the fuel list has all available fuel to select
## Actual behavior

The list is empty, the stonesmelter cannot work.

javaw_2021-11-17-003931-262

## Steps to reproduce the issue

1. 
2. 
3. 

### Logs

- latest.log: 
- crashlog: 

### Notes

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by uecasm**
Closes #7775

# Changes proposed in this pull request:
- Fix remote client tag sync/discovery race

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/event/TagWorkAroundEventHandler.java [hunk 1]
```diff
@@ -104,19 +108,56 @@ public static void onDataPackSync(final OnDatapackSyncEvent event)
...
+         * @param maybeRecipeManager null or a new RecipeManager to temporarily cache
+         */
+        private static void maybeLoadRecipes(@Nullable final RecipeManager maybeRecipeManager)
+        {
+            if (maybeRecipeManager != null)
+            {
+                recipeManager = maybeRecipeManager;
+            }
+
+            ++eventCounter;
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/event/TagWorkAroundEventHandler.java:129]: This needs to be reset on server leave.

Comment (uecasm) [reply src/main/java/com/minecolonies/coremod/event/TagWorkAroundEventHandler.java:129]: It's reset when it hits 2 (which should always happen unless they somehow disconnect between the two packets, in which case they presumably wouldn't get a leave event either?).  Or if something goes weird, that's why it checks for a non-null recipeManager too.

**Review by Raycoms — APPROVED** [review]

**Comment by marchermans** [comment]
Why was this merged?

**Comment by marchermans** [comment]
This has only a single approve, and is implemented really poorly?

**Comment by Raycoms** [comment]
Merged it because its a hotfix (affects gameplay significantly) and your only comment was on the reset on server leave which he commented on and I didn't find necessary either (as it resets on 2 anyway).

**Comment by uecasm** [comment]
I'm happy to make another PR that also adds a reset on logout if you like, but I don't think it's needed.  In the incredibly unlikely case that someone disconnects between receiving the two packets (that are sent immediately following each other), it will be self-correcting on the next login or reload.

**Pull request merged**